registries/adapters/galaxy_adapter.go

```diff
@@ -0,0 +1,259 @@
+	}
+	// if there was an issue with the first call, return the error
+	if err != nil {
+		return nil, err
+	}
+
+	// If no results in the fist call then close the channel as nothing will get loaded.
+	if len(imageResp.Results) == 0 {
+		log.Info("canceled retrieval as no items in org")
+		close(channel)
+	}
+	var apbData []string
+	counter := 1
+	for imageData := range channel {
+		apbData = append(apbData, imageData)
+		if counter < imageResp.Count {
+			counter++
+		} else {
+			close(channel)
+		}
+	}
+	// check to see if the context had an error
+	if ctx.Err() != nil {
+		log.Errorf("encountered an error while loading images, we may not have all the apb in the catalog - %v", ctx.Err())
+		return apbData, ctx.Err()
+	}
+
+	return apbData, nil
+}
+
+// FetchSpecs - retrieve the spec for the image names.
+func (r GalaxyAdapter) FetchSpecs(imageNames []string) ([]*bundle.Spec, error) {
+	specs := []*bundle.Spec{}
+	for _, imageName := range imageNames {
+		spec, err := r.loadSpec(imageName)
+		if err != nil {
+			log.Errorf("Failed to retrieve spec data for image %s - %v", imageName, err)
+		}
+		if spec != nil {
+			specs = append(specs, spec)
+		}
+	}
+	return specs, nil
+}
+
+// getNextImages - will follow the next URL using go routines.
+func (r GalaxyAdapter) getNextImages(ctx context.Context,
+	url string, ch chan<- string,
+	cancelFunc context.CancelFunc) (*GalaxySearchResponse, error) {
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		log.Errorf("unable to get next roles for url: %v - %v", url, err)
+		cancelFunc()
+		close(ch)
+		return nil, err
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Errorf("unable to get next roles for url: %v - %v", url, err)
+		cancelFunc()
+		close(ch)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	imageList, err := ioutil.ReadAll(resp.Body)
+
+	iResp := GalaxySearchResponse{}
+	err = json.Unmarshal(imageList, &iResp)
+	if err != nil {
+		log.Errorf("unable to get next images for url: %v - %v", url, err)
+		cancelFunc()
+		close(ch)
+		return &iResp, err
+	}
+	// Keep getting the images
+	if iResp.Next != "" {
+		log.Debugf("getting next page of results - %v", iResp.Next)
+		// Fan out calls to get the next images.
+		go r.getNextImages(ctx, fmt.Sprintf("%v%v", galaxyAPIURL, iResp.Next), ch, cancelFunc)
+	}
+	for _, imageName := range iResp.Results {
+		log.Debugf("Trying to load %v.%v", imageName.Summary.Namespace.Name, imageName.Name)
+		go func(image *GalaxyRole) {
+			select {
+			case <-ctx.Done():
+				log.Debugf(
+					"loading images failed due to context err - %v name - %v",
+					ctx.Err(), image.Name)
+				return
+			default:
+				ch <- fmt.Sprintf("%v.%v#%v", image.Summary.Namespace.Name, image.Name, image.RoleID)
+			}
+		}(imageName)
+	}
+	return &iResp, nil
+}
+
+func (r GalaxyAdapter) loadSpec(imageName string) (*bundle.Spec, error) {
+	imageSplit := strings.Split(imageName, "#")
+	roleName := imageSplit[0]
+	roleID := imageSplit[1]
```

Comment:
if `imageName` does not have a # in it, we will panic here. We should really check the length of `imageSplit`